a10_octavia/tests/unit/controller/worker/tasks/test_vthunder_tasks.py

```diff
@@ -0,0 +1,248 @@
+
+VTHUNDER = a10_data_models.VThunder()
+VLAN_ID = '11'
+VE_IP_VLAN_ID = '12'
+DELETE_VLAN = True
+TAG_INTERFACE = '5'
+ETH_DATA = {"action": "enable"}
+SUBNET_MASK = ["10.0.11.0", "255.255.255.0", "10.0.11.0/24"]
+VE_IP_SUBNET_MASK = ["10.0.12.0", "255.255.255.0", "10.0.12.0/24"]
+RACK_DEVICE = {
+    "project_id": "project-rack-vthunder",
+    "ip_address": "10.0.0.1",
+    "device_name": "rack_vthunder",
+    "username": "abc",
+    "password": "abc",
+    "interface_vlan_map": {"5": {"11": {"use_dhcp": "True"},
+                                 "12": {"ve_ip_address": ".10"}}}
+}
+DEL_VS_LIST = {"virtual-server-list": [{"ip-address": "10.0.1.1"}]}
+DEL_SERVER_LIST = {"server-list": [{"host": "10.0.2.1"}]}
+DEL_PORT_ID = "mock-port-id"
+VE_IP = "10.0.11.10"
+STATIC_VE_IP = "10.0.12.10"
+NETWORK_11 = n_data_models.Network(id="mock-network-1", subnets=["mock-subnet-1"],
+                                   provider_segmentation_id=11)
+NETWORK_12 = n_data_models.Network(id="mock-network-2", subnets=["mock-subnet-2"],
+                                   provider_segmentation_id=12)
+
+
+class TestVThunderTasks(base.BaseTaskTestCase):
+
+    def setUp(self):
+        super(TestVThunderTasks, self).setUp()
+        self.conf = self.useFixture(oslo_fixture.Config(cfg.CONF))
+        imp.reload(task)
+        self.client_mock = mock.Mock()
+
+    def tearDown(self):
+        super(TestVThunderTasks, self).tearDown()
+        self.conf.reset()
+
+    def _mock_lb(self):
+        lb = mock.Mock()
+        lb.id = a10constants.MOCK_LOAD_BALANCER_ID
+        lb.project_id = "project-rack-vthunder"
+        lb.vip.subnet_id = "mock-subnet-1"
+        return lb
+
+    def _mock_member(self):
+        member = mock.Mock()
+        member.id = a10constants.MOCK_MEMBER_ID
+        member.project_id = "project-rack-vthunder"
+        member.subnet_id = "mock-subnet-1"
+        return member
+
+    @mock.patch('keystoneauth1.session.Session', mock.Mock())
+    @mock.patch('neutronclient.client.SessionClient', mock.Mock())
+    @mock.patch('neutronclient.v2_0.client.Client', mock.Mock())
+    def _mock_tag_task(self, mock_task):
+        mock_task.axapi_client = self.client_mock
+        mock_task.get_vlan_id = mock.Mock()
+        mock_task.get_vlan_id.return_value = VLAN_ID
+        mock_task._get_ve_ip = mock.Mock()
+        mock_task._get_ve_ip.return_value = VE_IP
+        mock_task.get_subnet_and_mask = mock.Mock()
+        mock_task.get_subnet_and_mask.return_value = [SUBNET_MASK[0], SUBNET_MASK[1]]
+        mock_task._subnet_ip = SUBNET_MASK[0]
+        mock_task._subnet_mask = SUBNET_MASK[1]
+        mock_task._subnet = mock.Mock()
+        mock_task._subnet.cidr = SUBNET_MASK[2]
+        mock_task._subnet = mock.Mock()
+        mock_task._subnet.id = mock.Mock()
+        mock_task._subnet.network_id = "mock-network-1"
+        mock_task._network_driver = a10_octavia_neutron.A10OctaviaNeutronDriver()
+
+    def test_tag_ethernet_for_lb(self):
```

Comment:
We need to change these names to `test_ClassName/function_name_operation/expteced_outcome_input`

Reply:
Done.